utilities/post_migration.py

```
@@ -399,32 +399,136 @@ def _verify_gateway(interface_name: str, expected_gateway: str, matching_interfa
+    if not ssh_conn.rrmngmnt_host:
+        LOGGER.warning(f"SSH connection not established for VM {vm_name} - retrying...")
+        return None
+
+    executor = ssh_conn.rrmngmnt_host.executor(user=ssh_conn.rrmngmnt_user)
+    executor.port = ssh_conn.local_port
+    rc, stdout, err = executor.run_cmd(network_cmd)
+    if rc != 0:
+        raise CommandExecFailed(name=" ".join(network_cmd), err=err)
+
+    if not stdout.strip():
+        LOGGER.warning(f"{' '.join(network_cmd)} returned empty output")
```

Ruff (0.15.12)

[warning] Logging statement uses f-string

(G004)